subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/console/taskgrouping/PlainConsoleGroupedTaskFunctionalTest.groovy

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.logging.console.taskgrouping
+
+import groovy.transform.NotYetImplemented
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+import static org.gradle.api.logging.configuration.ConsoleOutput.Plain
+
+class PlainConsoleGroupedTaskFunctionalTest extends AbstractIntegrationSpec {
```

Comment:
we should reuse as many of the existing tests for the console grouping, rather than reimplementing them all over again. Pretty much all of the existing cases should also work with the plain console, with a little abstraction over how to find the task specific output in the process' stdout/stderr outputs.

The existing tests do need some refactoring to remove some assumptions (always use the rich console, always run with `--parallel`) in order to allow them to be reused. So, we could keep this test class for now so we can get things working and then come back to refactor at least enough so that there are no duplicated test cases.

Reply:
👍 That makes a lot of sense.